File: codes/ruby/chapter06/answer_A41.rb
# 入力
N = gets.chomp.to_i
S = gets.chomp

# 答えを求める
Answer = false
(0..N-3).each do |i|
  Answer = true if S[i] == 'R' && S[i + 1] == 'R' && S[i + 2] == 'R'
  Answer = true if S[i] == 'B' && S[i + 1] == 'B' && S[i + 2] == 'B'
end

# 出力
if Answer
  puts "Yes"
else
  puts "No"
end

File: codes/go/chapter06/answer_A41.go
package main

import "fmt"

func main() {
	var N int
	var S string
	fmt.Scan(&N, &S)

	// 答えを求める
	Answer := false
	for i := 0; i <= N-3; i++ {
		if S[i] == 'R' && S[i+1] == 'R' && S[i+2] == 'R' {
			Answer = true
		}
		if S[i] == 'B' && S[i+1] == 'B' && S[i+2] == 'B' {
			Answer = true
		}
	}

	// 出力
	if Answer == true {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}
